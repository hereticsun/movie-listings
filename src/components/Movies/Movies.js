import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchMovies } from '../../actions/movies';
import Movie from '../Movie';

import styles from './Movies.module.css';

export class Movies extends Component {
  componentDidMount() {
    this.props.fetchMovies();
  }

  render() {
    const moviesData = this.props.movies.moviesList;
    return (
      <section className={styles.movies}>
        <h2 className={styles.moviesTitle}>Now playing</h2>
        {Boolean(moviesData) && moviesData.length > 0 ? (
          <ul className={styles.moviesList}>
            {moviesData.map(movie => (
              <Movie movie={movie} key={movie.id} />
            ))}
          </ul>
        ) : (
          <p>No movies were found</p>
        )}
      </section>
    );
  }
}

function mapStateToProps(state) {
  return {
    movies: state.movies
  };
}

Movies.propTypes = {
  movies: PropTypes.object,
};

export default connect(mapStateToProps, { fetchMovies })(Movies);
