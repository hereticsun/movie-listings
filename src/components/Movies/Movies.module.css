.movies {
  max-width: 100%;
}

.moviesTitle {
  color: #fff;
  font-size: 24px;
}

.moviesList {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  margin: 0 auto;
  padding: 0;
}