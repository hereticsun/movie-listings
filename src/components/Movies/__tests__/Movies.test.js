import React from 'react';
import { shallow } from 'enzyme';
import { Movies } from '../Movies';

describe(`Movies`, () => {
  const props = {
    movies: {
      moviesList: [
        {
          id: 1,
          title: 'Toy Story 4',
        }
      ]
    },
    fetchMovies: jest.fn()
  };

  const element = shallow(<Movies {...props} />);

  it('displays the list of movies', () => {
    expect(element).toMatchSnapshot();
    expect(element.find('Movie')).toHaveLength(props.movies.moviesList.length);
  });

});
