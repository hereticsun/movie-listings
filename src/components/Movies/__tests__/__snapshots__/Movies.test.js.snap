// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies displays the list of movies 1`] = `ShallowWrapper {}`;
